tests/test_io.py

```diff
@@ -1690,3 +1691,378 @@ def test_csv_options_cs2(self):
+
+            Parameters:
+            ----------
+            dynamic_diagnostics : bool
+                Whether to enable dynamic diagnostics during simulation.
+
+            nbr_of_calls : int
+                Number of simulation calls to perform.
+
+            diag_data_ratio : int
+                Frequency of diagnostic data logging.
+
+            vars_to_test : list
+                Variables to retrieve data from during the simulation.
+
+            u : callable
+                Function to determine the stop index for retrieving variable data.
+
+            result_file_name: str
+                Name of the result file.
+
+            Returns:
+            -------
+            dict
+                A dictionary with call indices as keys and lists of trajectory data as values.
+
+            Raises:
+            ------
+            AssertionError
+                If no test data is generated.
+        """
+        fmu = Dummy_FMUModelME2([], os.path.join(file_path, "files", "FMUs", "XML", "ME2.0", "CoupledClutches.fmu"),
+            _connect_dll=False)
+
+        result_handler = ResultHandlerBinaryFile(fmu)
+        opts = fmu.simulate_options()
+        opts["result_handling"] = "binary"
+        opts["result_handler"] = result_handler
+        opts["result_file_name"] = result_file_name
+        opts["dynamic_diagnostics"] = dynamic_diagnostics
+        opts["logging"] = opts["dynamic_diagnostics"]
+
+        # tolerances are required only to invoke 'setup_diagnostics_variables'
+        opts['CVode_options']['rtol'] = 1e-6
+        opts['CVode_options']['atol'] = fmu.nominal_continuous_states * opts['CVode_options']['rtol']
+
+        fmu.setup_experiment()
+        fmu.initialize()
+        opts["initialize"] = False
+
+        diag_params, diag_vars = setup_diagnostics_variables(
+            model = fmu,
+            start_time = 0,
+            options = opts,
+            solver_options = opts['CVode_options'])
+
+        result_handler.set_options(opts) # required in order to call simulation_start()
+        result_handler.initialize_complete()
+        if opts["dynamic_diagnostics"]:
+            result_handler.simulation_start(diag_params, diag_vars)
+        else:
+            result_handler.simulation_start()
+
+        fmu.set('J4.phi', math.sin(5*fmu.time + math.pi/2)) # arbitrary
+        result_handler.integration_point()
+
+        fmu.time += 0.01
+        fmu.set('J4.phi', math.sin(5*fmu.time + math.pi/2)) # arbitrary
+
+        rdb = ResultDymolaBinary(opts["result_file_name"], allow_file_updates=True)
+        nbr_of_calls = nbr_of_calls
+        diag_data_ratio = diag_data_ratio
+        counter = 0
+        start_index = 0
+
+        nbr_of_points = 5
+        h = 1/nbr_of_points * 0.1 # "arbitrary" step through time
+
+        data_to_return = {}
+        for i in range(nbr_of_calls):
+            time.sleep(0.1) # without this the tests fail on Linux, should look into a proper way of solving this
```

Comment:
I don't like this line here at all, but it is necessary in order to have the tests pass on Linux. I am open to other ideas or leave this as a TODO/follow-up ticket for now.

The reason why this is necessary is because on Linux, in the following function:
```
def _verify_file_data(self):
    if self._mtime != os.path.getmtime(self._fname): # this doesn't evaluate to True on Linux, because os.path.getmtime does not update as 'expected' (compared to Windows)
```
```

Reply:
Maybe requires a follow-up to look into an alternative to verify file data that also works on linux?

Also, shouldn't the sleep be in between the integration/diagnostics point & getting the variable data?

Reply:
Yes I am planning to create a follow-up ticket, Ill do that as the last part of the review.
That is what I thought too, but the sleep between `*_point()` and getting the data doesn't help which makes this entire thing even more odd.

Reply:
What is it that drives the simulation and generates new data for `result_handler.integration_point()` to write? I would assume some fmiCompletedIntegratorStep, but don't see any, but maybe it's hidden away in the dummy FMU class.  Given that you need to sleep at the start of the loop, maybe this could be related?

Reply:
In this test case we don't simulate the `normal way`, since we dont use `.simulate()`. We step manually by invoking `integration_point`, manually advance the time, update one of the state variables etc.

Reply:
I was about to test one final thing but I can no longer reproduce the issue even when I remove the `sleep`. I wonder if this is caused by the tests that I removed in this commit https://github.com/modelon-community/PyFMI/pull/267/commits/a7524eff12b2454b70be32fc554cc9ceaee17753.

They are different tests but they are very similar, except that they dont check the returned values.

Reply:
I see now that the tests from github actions fail, which is different from my own Ubuntu machine. I will add `sleep` back...